### PR TITLE
Replaced demo with Kim's project

### PR DESCRIPTION
**Pull request by KimReth**
Created WebApi namespace within types for create/delete/update/retrieve record functionality.
Created new_OnLoad.JS.ts web resource.
Added additional Control functionality within typegen, autogenerates getControl.
Fixed bug within typegen that produced an undefined error within typegenForm.
Added refresh() to control.d.ts.

### CODE REVIEW

File: demo/xrm.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
     "auth": {
-        "authority": "https://login.microsoftonline.com/22f99e51-8625-4879-ba58-ea18fd4b299a",
-        "clientId": "4ce1950f-d9e0-453b-813f-90fd14ed4cc5"
+        "authority": "https://login.microsoftonline.com/ac2d7ce5-fae0-42e9-b01a-2eb6e86c0eab",
+        "clientId": "106f1a90-7645-46b8-b5fa-cf85de7d480b"
     },
-    "dynamics": "https://org3652e152.crm.dynamics.com",
+    "dynamics": "https://orgc840d94a.crm.dynamics.com",
```
Comment (Xisrith) [inline demo/xrm.json:6]: Please forward me the credentials to this, at some point I'll figure out a good way to store them for the team

File: types/index.d.ts [hunk 2]
```diff
@@ -163,5 +163,61 @@ declare global {
...
+            /**
+             * Creates a table record
+             * 
+             * See {@link https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-webapi/createrecord docs.microsoft.com}
+             * 
+             * @param entityLogicalName Logical name of the table you want to create. For example: "account"
+             * @param data A JSON object defining the columns and values for the new table record.
+             * 
+             */
+            function createRecord(entityLogicalName: string, data: any): Promise<any>;
```
Comment (Xisrith) [inline types/index.d.ts]: Please add an inline type for the promise return type

File: types/index.d.ts [hunk 3]
```diff
@@ -163,5 +163,61 @@ declare global {
...
+             * Retrieves a table record
+             * 
+             * See {@link https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-webapi/retrieverecord docs.microsoft.com}
+             * 
+             * @param entityLogicalName The table logical name of the record you want to retrieve. For example: "account".
+             * @param id GUID of the table record you want to retrieve.
+             * @param options OData system query options, $select and $expand, to retrieve your data.
+             * 
+             */
+            function retrieveRecord(entityLogicalName: string, id: string, options?: string): Promise<any>;
```
Comment (Xisrith) [inline types/index.d.ts:185]: Please add an inline type for the promise return type

File: types/index.d.ts [hunk 4]
```diff
@@ -163,5 +163,61 @@ declare global {
...
+            /**
+             * Deletes a table record
+             * 
+             * See {@link https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-webapi/deleterecord docs.microsoft.com}
+             * 
+             * @param entityLogicalName The table logical name of the record you want to delete. For example: "account".
+             * @param id GUID of the table record you want to delete.
+             * 
+             */
+            function deleteRecord(entityLogicalName: string, id: string): Promise<any>;
```
Comment (Xisrith) [inline types/index.d.ts]: Please add an inline type for the promise return type

File: types/index.d.ts [hunk 5]
```diff
@@ -163,5 +163,61 @@ declare global {
...
+             * Updates a table record
+             * 
+             * See {@link https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-webapi/updaterecord docs.microsoft.com}
+             * 
+             * @param entityLogicalName The table logical name of the record you want to update. For example: "account".
+             * @param id GUID of the table record you want to update.
+             * @param data A JSON object containing key: value pairs, where `key` is the property of the table and value is the value of the property you want to update.
+             * 
+             */
+            function updateRecord(entityLogicalName: string, id: string, data: any): Promise<any>;
```
Comment (Xisrith) [inline types/index.d.ts]: Please add an inline type for the promise return type

File: types/index.d.ts [hunk 6]
```diff
@@ -163,5 +163,61 @@ declare global {
...
+            /**
+             *
```
Comment (Xisrith) [inline types/index.d.ts]: Remove extra whitespace in comment block

**Review by Xisrith — CHANGES_REQUESTED** [review]
Looks good, couple of comments regarding the Xrm.WebApi types

File: types/index.d.ts (hunk 3 above)
Comment (KimReth) [reply types/index.d.ts:185]: https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/xrm-webapi/retrieverecord

Left as any due to return being a JSON object

File: types/index.d.ts (hunk 2 above)
Comment (KimReth) [reply types/index.d.ts]: Added

File: types/index.d.ts (hunk 4 above)
Comment (KimReth) [reply types/index.d.ts]: Added

File: types/index.d.ts (hunk 5 above)
Comment (KimReth) [reply types/index.d.ts]: Added

File: types/index.d.ts (hunk 6 above)
Comment (KimReth) [reply types/index.d.ts]: Removed whitespace

File: demo/xrm.json (hunk 1 above)
Comment (KimReth) [reply demo/xrm.json:6]: Sent

**Review by Xisrith — APPROVED** [review]

**Pull request merged**